storybookjs/storybook
# Maintenance: Split renderers preview entrypoints

## What I did

When doing research how to optimize storybook compile time (`sb build`), @kasperpeulen discovered that we add docs related preview code even if addon docs isn't used.

This PR splits the injected code into 2 parts and ensures the docs related part is only injected when addon docs is active.

I renamed the `config.ts` (old naming scheme) file to `entry-preview.ts` and `entry-docs.ts`.
At first I had `preview.ts` and `preview-docs.ts`, but I found it hard to distinguish between entries by addons that were also called `preview`.

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

- Create a sandbox using vite
- Remove addon docs.
	- remove `.mdx` extension for stories
	- remove `addon-essentials` (and optionally `addon-docs` too)
	- remove docs options (usually configured like: `autodocs: 'tag'`)
	- remove any renderer (copied) specific stories
- add `rollup-plugin-visualizer` in `main.ts`:
   ```ts
    import { defineConfig } from 'vite';
    import react from '@vitejs/plugin-react';

    import { visualizer } from 'rollup-plugin-visualizer';

    export default defineConfig({
      plugins: [
        react(),
        visualizer({
          template: 'sunburst', // or sunburst
          open: true,
          gzipSize: true,
          brotliSize: true,
          filename: 'analyse.html', // will be saved in project's root
        }),
      ],
    });
    ```
- Inspect the opened webpage detailing the bundle stats.
- There should be no docs related code there. specifically the entry called `"entry-docs"` should NOT be there.

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
